payments-core/src/main/java/com/stripe/android/payments/core/injection/PaymentCommonModule.kt

```diff
@@ -55,10 +57,20 @@ abstract class PaymentCommonModule {
         @Singleton
         internal fun provideStripeApiRepository(
```

Comment:
Since `PaymentCommonMoudle` is only used in `PaymentSheet`,(`FlowControllerComponent` and `PaymentSheetViewModelComponent`), it could be combined with `PaymentSheetCommonModule`.

Now that we already have [StripeRepositoryModule](https://github.com/stripe/stripe-android/blob/b6fdd9280905b39f60af9dde9d69bbb97568ab67/payments-core/src/main/java/com/stripe/android/payments/core/injection/StripeRepositoryModule.kt#L24) providing `StripeRepository`, I think we could do the following:
1. remove `PaymentCommonModule`
2. move `PaymentCommonModule#providePaymentConfiguration` to `PaymentSheetCommonModule`
3. Let `FlowControllerComponent` and `PaymentSheetViewModelComponent` include `StripeRepositoryModule`

wdyt?

Reply:
What about `AnalyticsRequestExecutor` and `AnalyticsRequestFactory` which are also provided by `PaymentCommonMoudle`?

Reply:
They come with [StripeRepositoryModule](https://github.com/stripe/stripe-android/blob/b6fdd9280905b39f60af9dde9d69bbb97568ab67/payments-core/src/main/java/com/stripe/android/payments/core/injection/StripeRepositoryModule.kt#L24)

Reply:
Sounds good, but there are bigger and unrelated changes there, so I'll do in a separate PR.